apps/emqx_authn/src/simple_authn/emqx_authn_http.erl

```diff
@@ -261,21 +263,47 @@ transform_header_name(Headers) ->
     ).
 
 check_ssl_opts(Conf) ->
-    {BaseUrl, _Path, _Query} = parse_url(get_conf_val("url", Conf)),
-    case BaseUrl of
-        <<"https://", _/binary>> ->
-            case get_conf_val("ssl.enable", Conf) of
-                true -> ok;
-                false -> false
+    case is_backend_http(Conf) of
+        true ->
+            Url = get_conf_val("url", Conf),
+            {BaseUrl, _Path, _Query} = parse_url(Url),
+            case BaseUrl of
+                <<"https://", _/binary>> ->
+                    case get_conf_val("ssl.enable", Conf) of
+                        true ->
+                            ok;
+                        false ->
+                            <<"it's required to enable the TLS option to establish a https connection">>
+                    end;
+                <<"http://", _/binary>> ->
+                    ok
             end;
-        <<"http://", _/binary>> ->
+        false ->
             ok
     end.
 
 check_headers(Conf) ->
-    Method = to_bin(get_conf_val("method", Conf)),
-    Headers = get_conf_val("headers", Conf),
-    Method =:= <<"post">> orelse (not maps:is_key(<<"content-type">>, Headers)).
+    case is_backend_http(Conf) of
+        true ->
+            Headers = get_conf_val("headers", Conf),
+            case to_bin(get_conf_val("method", Conf)) of
+                <<"post">> ->
+                    ok;
+                <<"get">> ->
+                    case maps:is_key(<<"content-type">>, Headers) of
```

Comment:
we should add a converter to just drop this header if provided.

Reply:
Why we drop headers silently ? 
Maybe we should tell user, we don't carry this header, otherwise it's hard to find problem.

Reply:
if this was added from day one, then it's probably not an issue,
but we are now adding a more strict check on more lossy check older version. I'm not even sure if node will successfully boot after this change.

Reply:
I'm sure this change is ok for older version.
The older version always check config when fill default. 
https://github.com/emqx/emqx/blob/master/apps/emqx_authn/src/emqx_authn_api.erl#L1133-L1136
this check_config alway call those two validators.